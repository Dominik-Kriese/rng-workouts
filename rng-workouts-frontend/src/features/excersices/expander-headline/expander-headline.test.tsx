import React from 'react';
import {mount, shallow} from 'enzyme';
import 'jest-styled-components';
import ExpanderHeadline from './expander-headline';

describe('Expander Headline', () => {
   it('renders the headline in collapsed mode', () => {
      const callback = jest.fn();
       const headline = shallow(<ExpanderHeadline title={'title'} expanded={false} callback={callback}/>)
           .dive()
           .find('h2');
       expect(headline).toMatchSnapshot();
   });

   it('renders the headline in expanded mode', () => {
       const callback = jest.fn();
       const headline = shallow(<ExpanderHeadline title={'title'} expanded={true} callback={callback}/>)
           .dive()
           .find('h2');
       expect(headline).toMatchSnapshot();
   });

   test('calls the callback function on click', () => {
       const callback = jest.fn();
       const headline = shallow(<ExpanderHeadline title={'title'} expanded={true} callback={callback}/>)
           .dive()
           .find('h2');
       headline.simulate('click');
       expect(callback).toHaveBeenCalled();
   });
});
