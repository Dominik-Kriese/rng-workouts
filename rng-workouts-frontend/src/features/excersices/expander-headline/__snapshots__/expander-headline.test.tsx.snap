// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expander Headline renders the headline in collapsed mode 1`] = `
.c0 {
  font-weight: bold;
  font-size: 1.125rem;
}

<h2
  className="c0"
  onClick={[Function]}
>
  title
  <styled.span>
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "&ndash;",
        }
      }
    />
  </styled.span>
</h2>
`;

exports[`Expander Headline renders the headline in expanded mode 1`] = `
.c0 {
  font-weight: bold;
  font-size: 1.125rem;
}

<h2
  className="c0"
  onClick={[Function]}
>
  title
  <styled.span>
    +
  </styled.span>
</h2>
`;
